### PR TITLE
[server] API: Ignore write-after-close error

### PR DESCRIPTION
**Pull request by geropl**
## Description

I can't reproduce this in the preview, so not :100: sure this is enough to fix the issue. But the change is good anyway, and safe to deploy. :+1: 

## Related Issue(s)

Fixes ENT-232

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - gpl-232-waf5b9e82d01</li>
	<li><b>🔗 URL</b> - <a href="https://gpl-232-waf5b9e82d01.preview.gitpod-dev.com/workspaces" target="_blank">gpl-232-waf5b9e82d01.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - gpl-232-watch-prebuild-gha.25791</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-gpl-232-waf5b9e82d01%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Comment by geropl** [comment]
@WVerlaek This should fix the issues we are seeing :crossed_fingers:

File: components/server/src/api/prebuild-service-api.ts [hunk 1]
```diff
@@ -124,6 +124,12 @@ export class PrebuildServiceAPI implements ServiceImpl<typeof PrebuildServiceInt
         const it = this.prebuildManager.getAndWatchPrebuildStatus(ctxUserId(), filter, { signal: ctxSignal() });
 
         for await (const pb of it) {
+            if (ctxSignal().aborted) {
```
Comment (mustard-mh) [inline components/server/src/api/prebuild-service-api.ts:127]: I thought they will be cancelled together here https://github.com/gitpod-io/gitpod/blob/097e2c3c29b085ce99ecc34c4b4a7d5fc2e0a1d0/components/gitpod-protocol/src/generate-async-generator.ts#L27

But not sure if it's actually working

Comment (mustard-mh) [reply components/server/src/api/prebuild-service-api.ts:127]: Should work? because we have test cases for  this function

Comment (mustard-mh) [reply components/server/src/api/prebuild-service-api.ts:127]: @geropl  see also https://github.com/connectrpc/connect-es/blob/v1.1.2/packages/connect-node/src/node-universal-handler.ts#L227-L241

Comment (mustard-mh) [reply components/server/src/api/prebuild-service-api.ts:127]: > Should we try to catch that error and silence it? WDYT? 🤔

👍

File: components/server/src/api/server.ts [hunk 2]
```diff
@@ -223,6 +223,20 @@ export class API {
                     };
                     const handleError = (reason: unknown) => {
                         const err = self.apiConverter.toError(reason);
+
+                        if (
+                            err.code === Code.Internal &&
```
Comment (geropl) [inline components/server/src/api/server.ts:228]: @mustard-mh Could you take a look? :pray:

**Review by filiptronicek — APPROVED** [review]

**Pull request merged**